docs/tasks/argo.rst

```diff
@@ -0,0 +1,253 @@
+Leveraging ActionSet Creation using Argo Cron Workflows
+------------------------------------------------------------
+
+Argo Workflows enable us to schedule operations. In the Kanister project,
+Argo Cron Workflows will be used to automate the creation of ActionSets to
+backup an application at regular intervals.
+
+To summarize, ActionSets are CRDs that are used to backup or restore an application.
+The controller present in the Kanister namespace looks for ActionSets to
+execute in the given namespace.
+
+In this tutorial you will schedule the creation of a backup actionset using
+Argo Cron Workflows.
+
+
+Prerequisites
+===============
+- Kubernetes 1.16+
+- Kanister controller version 0.80.0 installed in your cluster, let's assume in
+  Namespace kanister
+- Kanctl CLI installed (https://docs.kanister.io/tooling.html#install-the-tools)
+
+Architecture
+===============
+
+.. image:: img/argo-cron-architecture.png
+
+Steps -
+===============
+
+
+Step 1 - Setup Argo
+``````````````````````
+
+Download the Argo CLI from their Releases_ page.
+
+Create a separate namespace for the Workflows.
+
+.. code-block:: bash
+
+  kubectl create ns argo
+
+For this tutorial, the minimal manifest file will be used to download
+Argo CRDs and setup their resources.
+Download the minimal manifest file from their Examples_ page.
+
+On downloading this yaml file, you may configure the managed namespaces
+as per your use case.
+You may read more about managed namespaces here - ManagedNamespaces_
+
+In this tutorial, the minimal manifest will be used as it is.
+
+.. code-block:: bash
+
+  kubectl apply -f quick-start-minimal.yaml -n argo
+
+Port-forward to view the Argo UI:
+
+.. code-block:: bash
+
+  kubectl -n argo port-forward deployment/argo-server 2746:2746
+
+Open a web browser and navigate to ``localhost:2746``
+
+.. image:: img/argo-default-ui.png
+
+Step 2 - Setup a sample application to backup.
+``````````````````````````````````````````````````
+
+Here, you will reference the MySQL_ example from Kanister.
+Install the Chart and setup MySQL in the ``mysql-test`` namespace.
+Make sure that the application is integrated with Kanister and the following
+steps are completed.
```

Comment:
I think we should form proper sentences here, if possible, instead of bullet point sentences. For example.

We are going see how we can backup MySQL application by creating backup ActionSet. The details about the application can also be found in examples[link].

Reply:
I was trying to do 2 things here -
1. Avoid using pronouns like "We" as mentioned [here](https://github.com/kanisterio/kanister/blob/0c3c45d2d6a82b24169f85ebd0b7de164a754f2a/docs/README.md) 
2. Instruct the user to go to the link and follow the steps mentioned specifically as "Install the chart" and "Integrating with Kanister". Which is why the specific instruction to reference the document.

Reply:
I think in that case we can specifically mention that Setup MySQL application by going to this link and following those two points.

Reply:
Yes I was intending to do that but what if the MySQL documentation changes. Which is why I mentioned it in general.

Reply:
Done.